### PR TITLE
PR: Update macOS app with new dependencies

### PR DESCRIPTION
**Pull request by mrclary**
## Description of Changes
Fixed a few bugs and updated build requirements

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

* `setuptools` and `pyobjc` recent releases are now compatible with the macOS application build so the environment constraints file was updated accordingly.
* A bug was fixed in which `pyls_black`, `pyls_spyder`, and `setuptools` were not installed with the bundle. These packages were recently added requirements that seem to be undetected by `py2app`

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:
mrclary

### CODE REVIEW

**Review by ccordoba12 — APPROVED** [review]
Thanks @mrclary, looks good to me.

File: installers/macOS/setup.py [hunk 1]
```diff
@@ -100,24 +100,31 @@ def make_app_bundle(dist_dir, make_lite=False):
...
    """
     build_type = 'lite' if make_lite else 'full'
     logger.info('Creating %s app bundle...', build_type)
 
     PACKAGES = ['alabaster', 'astroid', 'ipykernel', 'IPython', 'jedi',
-                'jinja2', 'keyring', 'parso', 'pygments', 'pyls', 'qtawesome',
-                'spyder', 'spyder_kernels', 'sphinx']
+                'jinja2', 'keyring', 'parso', 'pygments', 'pyls', 'pyls_black',
+                'pyls_spyder', 'qtawesome', 'setuptools', 'sphinx', 'spyder',
+                'spyder_kernels']
```
Comment (ccordoba12) [inline installers/macOS/setup.py:127]: You're missing `three-merge` here, which is also a new dependency for 4.2

**Review by ccordoba12 — COMMENTED** [review]
Forgot on thing when I was about to merge.

File: installers/macOS/setup.py (hunk 1 above)
Comment (mrclary) [reply installers/macOS/setup.py:127]: I confirmed that `three-merge` is discovered by py2app and is included in the macOS app, so no further action is required.
Screen Shot 2020-11-03 at 1 47 14 PM

Comment (ccordoba12) [reply installers/macOS/setup.py:127]: Ok, thanks!

**Review by ccordoba12 — APPROVED** [review]
@mrclary confirmed `three-merge` is part of the app, so no further action is required.

**Pull request merged**